.build/build-sonar.xml

```diff
@@ -0,0 +1,167 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one
+  ~ or more contributor license agreements.  See the NOTICE file
+  ~ distributed with this work for additional information
+  ~ regarding copyright ownership.  The ASF licenses this file
+  ~ to you under the Apache License, Version 2.0 (the
+  ~ "License"); you may not use this file except in compliance
+  ~ with the License.  You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<project basedir="." name="apache-cassandra-sonar-tasks" xmlns:if="ant:if" xmlns:unless="ant:unless">
+    <property name="sonar-scanner-cli.version" value="5.0.1.3006-linux"/>
+    <property name="sonar.workdir" value="${build.dir}/sonar"/>
+    <property name="sonar.download.dir" value="${user.home}/.cassandra-cache/tools"/>
+    <property name="sonar-scanner.home" value="${sonar.download.dir}/sonar-scanner-${sonar-scanner-cli.version}"/>
+    <property name="sonar.projectKey" value="cassandra"/>
+    <property name="sonar.mainBranch" value="trunk"/>
+
+    <target name="sonar-init">
+        <mkdir dir="${sonar.download.dir}"/>
+        <mkdir dir="${sonar-scanner.home}"/>
+        <mkdir dir="${sonar.workdir}"/>
+        <mkdir dir="${sonar.workdir}/scanner"/>
+        <mkdir dir="${sonar.workdir}/data"/>
+        <mkdir dir="${sonar.workdir}/logs"/>
+        <mkdir dir="${sonar.workdir}/extensions"/>
+    </target>
+
+    <target name="init-sonar-scanner-cli" depends="sonar-init" unless="sonar-scanner.downloaded">
+        <local name="sonar-scanner.downloaded"/>
+        <available file="${sonar-scanner.home}/bin/sonar-scanner" property="sonar-scanner.downloaded"/>
+
+        <sequential unless:set="sonar-scanner.downloaded">
+            <echo>Downloading Sonar-Scanner CLI...</echo>
+            <retry retrycount="3" retrydelay="5000" >
+                <get src="https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-${sonar-scanner-cli.version}.zip"
+                     dest="${sonar.download.dir}/sonar-scanner-cli-${sonar-scanner-cli.version}.zip"/>
+            </retry>
+            <unzip src="${sonar.download.dir}/sonar-scanner-cli-${sonar-scanner-cli.version}.zip" dest="${sonar.download.dir}" overwrite="true"/>
+            <setpermissions mode="755">
+                <!-- unzip task in Ant does not preserve file permissions, so we need to make those files executable explicitly -->
+                <file file="${sonar-scanner.home}/bin/sonar-scanner" />
+                <file file="${sonar-scanner.home}/jre/bin/java" />
```

Comment:
I've got the following during my first run, the directory seems to have the correct permissions.

```
m@MBP cassandra % ls -la ~/.cassandra-cache/tools/sonar-scanner-5.0.1.3006-linux/jre/bin/java 
-rwxr-xr-x@ 1 m  s  16320 Apr 18 21:49 /Users/m/.cassandra-cache/tools/sonar-scanner-5.0.1.3006-linux/jre/bin/java
```

```
init-sonar-server:
 [exec] UP

sonar:
 [exec] /Users/m/.cassandra-cache/tools/sonar-scanner-5.0.1.3006-linux/bin/sonar-scanner: line 66: /Users/m/.cassandra-cache/tools/sonar-scanner-5.0.1.3006-linux/jre/bin/java: cannot execute binary file
 [exec] /Users/m/.cassandra-cache/tools/sonar-scanner-5.0.1.3006-linux/bin/sonar-scanner: line 66: /Users/m/.cassandra-cache/tools/sonar-scanner-5.0.1.3006-linux/jre/bin/java: Undefined error: 0
 [exec] Result: 1

BUILD SUCCESSFUL
Total time: 2 minutes 20 seconds
m@MBP cassandra % 
```

Reply:
linux version of sonar...

Reply:
Note that I decided for the platform specific version, that is the one with attached JVM because the Scanner requires j17 and it is not obvious it is available in all the environments; it would be much stronger requirement than installing `jq`.